CairoMakie/src/primitives.jl

```diff
@@ -287,32 +267,32 @@ function draw_marker(ctx, marker::Char, font, pos, scale, strokecolor, strokewid
     cairo_font_face_destroy(cairoface)
 
     set_font_matrix(ctx, old_matrix)
-    Cairo.restore(ctx)
-
 end
 
+function draw_marker(ctx, marker::Circle, pos, scale, strokecolor, strokewidth, marker_offset, rotation)
 
-function draw_marker(ctx, marker::Union{Rect, Type{<: Rect}}, pos, scale, strokecolor, strokewidth, marker_offset, rotation)
-    s2 = if marker isa Type{Rect}
-        Point2(scale[1], -scale[2])
-    else
-        Point2((widths(marker) .* scale .* (1, -1))...)
-    end
+    marker_offset = marker_offset + scale ./ 2
+    pos += Point2f0(marker_offset[1], -marker_offset[2])
+    Cairo.arc(ctx, pos[1], pos[2], scale[1]/2, 0, 2*pi)
```

Comment:
So for Circle and Rect to be the same apparent size at the same markersize, the scale shouldn't be the diameter of the circle and the side length of the rect. I think if you go with the area approach it becomes clear that rect should be 1 side length for area 1, and circle should be whatever you need there, don't know from the top of my head but calculated it for a bunch of shapes in the bezierpath branch.

Reply:
I think we should do any changes like this in a new PR ;)